internal/csfle_util.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) MongoDB, Inc. 2022-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package internal
+
+import (
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+type StateCollection uint8
+
+const (
+	EncryptedCacheCollection StateCollection = iota
+	EncryptedStateCollection
+	EncryptedCompactionCollection
+)
```

Comment:
Optional: Is there a reason to define enum-ish constants? If not, consider simplifying this code by using string constants instead.

E.g.
```go
const (
	EncryptedCacheCollection      = "ecc"
	EncryptedStateCollection      = "esc"
	EncryptedCompactionCollection = "ecoc"
)
```
or
```go
type StateCollection string

const (
	EncryptedCacheCollection      StateCollection = "ecc"
	EncryptedStateCollection      StateCollection = "esc"
	EncryptedCompactionCollection StateCollection = "ecoc"
)
```

Reply:
I like the idea of string constants. Done.